### PR TITLE
PRO-4150: Deceased DoB

### PR DESCRIPTION
**Pull request by Wivenhoe-HMCTS**
add deceased date of birth to caveat journey

### CODE REVIEW

File: app/resources/en/translation/deceased/dobknown.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "title": "Deceased date of birth",
+  "question": "Do you know {deceasedName}'s date of birth?",
```
Comment (LucaDelBuonoHMCTS) [inline app/resources/en/translation/deceased/dobknown.json]: Replace the single quote with `&rsquo;`

File: app/steps/ui/deceased/dobknown/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+class DeceasedDobKnown extends ValidationStep {
+
+    static getUrl() {
+        return '/deceased-dob-known';
+    }
+
+    nextStepUrl(ctx) {
+        return this.next(ctx).constructor.getUrl('noWill');
+    }
```
Comment (LucaDelBuonoHMCTS) [inline app/steps/ui/deceased/dobknown/index.js]: I don't think you'll need this method.

File: test/component/deceased/testDoB.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            testWrapper.testErrors(done, data, 'invalid', errorsToTest);
+
+        });
+
+        // it(`test it redirects to deceased domiciled: ${expectedNextUrlForDeceasedDomicile}`, (done) => {
+        //     const data = {dob_day: '01', dob_month: '01', dob_year: '1945'};
+        //     testWrapper.testRedirect(done, data, expectedNextUrlForDeceasedDomicile);
+        // });
+    });
+});
```
Comment (LucaDelBuonoHMCTS) [inline test/component/deceased/testDoB.js]: We have more tests in probate, duplicate that file.

File: test/unit/testDeceasedDob.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+                    msg: {
+                        summary: content.errors.dob_date.dateInFuture.summary,
+                        message: content.errors.dob_date.dateInFuture.message
+                    }
+                }
+            ]);
+            done();
+        });
+
+        it('should return the error for a date in the future', (done) => {
```
Comment (LucaDelBuonoHMCTS) [inline test/unit/testDeceasedDob.js]: Same test as above?

File: app/resources/en/translation/deceased/dobknown.json (hunk 1 above)
Comment (Wivenhoe-HMCTS) [reply app/resources/en/translation/deceased/dobknown.json]: changed

Comment (Wivenhoe-HMCTS) [reply app/resources/en/translation/deceased/dobknown.json]: to &apos;

File: test/unit/testDeceasedDob.js (hunk 4 above)
Comment (Wivenhoe-HMCTS) [reply test/unit/testDeceasedDob.js]: removed

File: app/resources/en/translation/deceased/dobknown.json (hunk 1 above)
Comment (LucaDelBuonoHMCTS) [reply app/resources/en/translation/deceased/dobknown.json]: Although technically it's an apostrophe, you're right, in Probate we've always used `&rsquo;` rather than `&apos;` ...

File: app/steps/ui/deceased/dobknown/index.js (hunk 2 above)
Comment (Wivenhoe-HMCTS) [reply app/steps/ui/deceased/dobknown/index.js]: removed

File: test/component/deceased/testDoB.js (hunk 3 above)
Comment (Wivenhoe-HMCTS) [reply test/component/deceased/testDoB.js]: file copied and updated

File: app/resources/en/translation/deceased/dobknown.json (hunk 1 above)
Comment (Wivenhoe-HMCTS) [reply app/resources/en/translation/deceased/dobknown.json]: changed to right ’ to keep in sync with probate

File: app/steps/ui/deceased/dobknown/index.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        };
+        return nextStepOptions;
+    }
+
+    generateContent(ctx, formdata) {
+        const content = super.generateContent(ctx, formdata);
+        const deceasedName = FormatName.format(formdata.deceased);
+        content.question = content.question.replace('{deceasedName}', deceasedName);
+        return content;
+    }
```
Comment (LucaDelBuonoHMCTS) [inline app/steps/ui/deceased/dobknown/index.js]: Replace this whole method with:
```
    getContextData(req) {
        const ctx = super.getContextData(req);
        const formdata = req.session.form;
        ctx.deceasedName = FormatName.format(formdata.deceased);
        return ctx;
    }
```

Comment (LucaDelBuonoHMCTS) [reply app/steps/ui/deceased/dobknown/index.js]: And update unit test accordingly

File: test/component/deceased/testDoDKnown.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        });
+
+        it(`test it redirects to deceased dob not known: ${expectedNextUrlForEndOfJourney}`, (done) => {
+            const data = {
+                dobknown: 'No'
+            };
+
+            testWrapper.testRedirect(done, data, expectedNextUrlForEndOfJourney);
+        });
+
```
Comment (LucaDelBuonoHMCTS) [inline test/component/deceased/testDoDKnown.js]: Add these tests:
```
        it('test right content loaded on the page', (done) => {
            testWrapper.testContent(done, []);
        });

        it('test errors message displayed for missing data', (done) => {
            const data = {};

            testWrapper.testErrors(done, data, 'required', []);
        });
```

File: app/steps/ui/deceased/dobknown/index.js (hunk 5 above)
Comment (Wivenhoe-HMCTS) [reply app/steps/ui/deceased/dobknown/index.js]: @shepda In conversation with David on Friday he confirmed that he would rather use the index.js file to replace the content variable that needs to be displayed with an appropriate unit test.

File: test/component/deceased/testDoDKnown.js (hunk 6 above)
Comment (Wivenhoe-HMCTS) [reply test/component/deceased/testDoDKnown.js]: updated

File: test/component/deceased/testDoDKnown.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        nock.cleanAll();
+    });
+
+    describe('Verify Content, Errors and Redirection', () => {
+        testHelpBlockContent.runTest('DeceasedDobKnown');
+
+        it('test right content loaded on the page', (done) => {
+            const contentData = {deceasedName: ''};
+            testWrapper.testContent(done, [], contentData);
+        });
```
Comment (LucaDelBuonoHMCTS) [inline test/component/deceased/testDoDKnown.js]: This needs to be more like this:
```
        it('test right content loaded on the page', (done) => {
            const sessionData = {
                deceased: {
                    firstName: 'John',
                    lastName: 'Doe'
                }
            };

            testWrapper.agent.post('/prepare-session/form')
                .send(sessionData)
                .end(() => {

                    const contentData = {deceasedName: 'John Doe'};

                    testWrapper.testContent(done, [], contentData);
                });
        });
```

Comment (Wivenhoe-HMCTS) [reply test/component/deceased/testDoDKnown.js]: test updated
